### DISCUSSION TITLE
Flickering DarkMode

### DISCUSSION THREAD

**Discussion by danilo-leal** [tailwindlabs/tailwindcss#3095] [Help] [answered]
Hey guys!

I've been migrating my personal portfolio from Webflow to Next.js with Tailwind.
Since Next.js does that server side rendering, I had to make a workaround to use localStorage in order to set the dark mode. 

But it's flickering whenever I change routes. Does anyone knows how to get around this? I've seen some stuff at the Next.js discussion space but I'm wondering if with the Tailwind approach to it there isn't any other way. 

Does anyone have any clue about it?
Check the site deployed at Vercel here: https://dl-co.vercel.app

**Comment by simonswiss** [comment]
Hey!

Yeah, it's kinda tricky! Here's a good thread you might find interesting and will get you on the right path:

https://github.com/vercel/next.js/discussions/12533

Would you be okay to share the source code of your project too?

**Reply by danilo-leal** [reply]
I red just that thread. But since I'm not so much of a pro at coding and the dude there isn't using Tailwind, I figured I might get a more objective help over here. I checked his code but couldn't think of a way to insert it into mine. But yea, of course! You can check out my website repository, it's public! Here's the function toggling the themes: https://github.com/danilo-leal/daniloleal.co/blob/main/components/atoms/DarkModeToggle.js

**Comment by danilo-leal** [comment, marked as answer]
Just figured this out using the incredible https://github.com/pacocoursey/next-themes abstraction. It worked like a charm. Too easy. I highly recommend to anyone using the Tailwind/Next combo.

**Comment by wizz-e** [comment]
i have the same issue but for vite

**Comment by mattslight** [comment]
If you are using Flowbite (this search result was showing as top for me) then you can use the solution noted here...

https://www.flowbite-react.com/docs/customize/dark-mode

> 
> NextJS[#](https://www.flowbite-react.com/docs/customize/dark-mode#nextjs)
> In server side rendered applications, such as NextJS, to avoid page flicker (if dark mode is set) before NextJS hydrates the content, ThemeModeScript component must be rendered in Head component (see implementation below).
> 
> ThemeModeScript renders a script tag that sets dark or removes dark from <html> element before hydration occurs.

```
// App router[#](https://www.flowbite-react.com/docs/customize/dark-mode#app-router)
// app/layout.tsx

import { ThemeModeScript } from 'flowbite-react';
 
export default function RootLayout({ children }) {
   return (
     <html>
       <head>
         <ThemeModeScript />
       </head>
       <body>{children}</body>
     </html>
   );
 }
```
